### ISSUE TITLE
Wire unusably slow in latest version

### ISSUE THREAD

**Issue by charliegreen** [wireapp/wire-android#1928]
## The problem

As of 3.24.681, Wire on Android has become almost impossible for me to use because it's so slow.

## Environment

* Wire version that exhibits the issue: 3.24.681
* Last Wire version that did not exhibit the issue (if applicable): 3.24
* Android OS version used to run Wire: 5.1
* Is this a custom firmware or a stock one: Stock
* Mobile phone model/manufacturer: Motorola Moto X (XT1058)
* Mobile network type (EDGE/LTE/Wi-Fi/Offline): Any

## Details

Wire is extremely slow. I can start typing and words won't appear for 10-20 seconds, when they all appear at once. I very frequently get OS pop-up dialogs asking if I want to wait for Wire to respond or if I want to kill it. Scrolling up and down can take an extremely long time as well, especially when the keyboard has to be hidden.

Sometimes, though, Wire will suddenly start working normally, and be completely usable and responsive for about 5-30 seconds, when it starts lagging again. I haven't seen any pattern in when it does this.

The chat where this is worst has an extremely long log, so I think this may be related to #1291; I figured that this deserved a new issue, though, because of how much worse it suddenly got in the latest version, and because it happens while typing as well as while sending messages and loading chats. The problem appears to be much less pronounced in chats with shorter logs.

I'd provide debug logs, but I'm afraid I can't because of #1927.

Thanks for your help!

**Comment by thefierybreeze** [comment]
Same issue for me and gf
We have a long chat log, it might be relevant.
Kind of emberrasing that this kind of bug came out after i switched so many friends over.
We are both running 7.1 lineageos, im on microG, she is on gapps, she has a slower phone keeps getting the wire is not responding notices

Sometimes when sending messages it doesn't get cleared instantly and you send it again only to end up sending two ditto messages, it's probably related.

I bet this has something to do with the new read receipts, the devs know better of course.

**Comment by ekianjo** [comment]
Same here. Wire has become unbearably slow in its latest version on Android. What is going on???

**Comment by marcoconti83** [comment]
Hi everyone and thanks for starting the conversation. 
Can you check if this is still the case with the latest public build? It should have a fix for this.

**Comment by alien2003** [comment]
Fixed for me

**Comment by ekianjo** [comment]
somewhat improved but not fixed. Now Wire hangs when trying to type until the previous message is confirmed as "delivered" which always takes several seconds.

**Comment by marcoconti83** [comment]
We are in the process of enabling more logs on public builds, so that user can send us log to investigate issues that we can't reproduce on our side. Since logs might contain sensitive information, we are performing an audit of the logs logic to verify that the logs are reasonably anonymized and contain no sensitive information. Once this is done and deployed, we will ask you to send logs which will help us debug this issue.

**Comment by ekianjo** [comment]
sounds good, will gladly provide logs if it helps.

**Comment by marcoconti83** [comment]
Hi everyone,
logs have been released to the public. 

This is how it works: before you can send logs, you need to enable logs recording. This is because we don't want to store this log file on your filesystem unless you decided to do it. As a consequence, you need to first enable logging, then reproduce your bug. If you do enable the logs after the bug has occurred, it will not contain the necessary information as nothing was recorded at that time.

To enable the logs, you can go to "Settings" -> "Advanced". You turn the logs on with the "Collect Usage Data" toggle. Once the logs are enabled, a log file is stored locally on your device. The log is not sent to Wire until you click on "Create debug report" in the same screen. When this happen, you are prompted to save/send the log file.

You can have a look at the log file yourself to make sure of what it contains; we have put an effort to anonymize most data via hashing. Once you are satisfied with the content, you can send it to our support (support@wire.com, and please mention the github ticket) or paste/link the content here, your choice.

**Comment by marcoconti83** [comment]
Hi everyone,
is this still an issue, or did you see some improvement?

**Comment by ekianjo** [comment]
I changed phone so the issue went away for me in a sense, however I face
the issue that no notifications display, even if Wire has battery saving
exceptions, push token rests and connectivity with the server turned on in
the settings. Was this ever reported?

On Monday, September 2, 2019, Marco <notifications@github.com> wrote:
> Hi everyone,
> is this still an issue, or did you see some improvement?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or mute the thread.<
https://ci4.googleusercontent.com/proxy/fGrrEDHBC-G_LnxntVmQ8X7CBnlOC3lAFyAN0agYT5yg0Ybi87cM2deiKmJcuoJ1yxAHKOXGirCYVj9F2dIRmJe5LHM74y8BZjMeYuUUEYik9dele1Q8tKwFY9fgluOJSOOBgRpXX0Kkp3cLRGCrMpUYW6VGL-N0uk74kJmQLmzY7HBIlRKS1ejJ97WgcZE4jk0HNdkkKV2CVasOAZuWDCkNUWFB6UvNlzvhnPiyJA=s0-d-e1-ft#https://github.com/notifications/beacon/AASYZGE5SMQZPXCSYHYVHT3QHTHATA5CNFSM4GMH3ZM2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD5VEX7A.gif
>

-- 
Raphael

**Comment by agates** [comment]
I'm experiencing this as well.  I've emailed some logs to support.

**Comment by Gardosen** [comment]
Hey @agates ,

i grabbed the log file from you via our customer support. they will get back to you with some questions, we will continue the communication regarding your issue via the ticket.
As this issue was originally related to 3.24, i will close this issue for now. 

and it will appear in the wild again, a new ticket can be created. but for clean up purpose and getting a better overview, i will close this for now.

thanks for your help with the debug log, i hope we can nail down the issue which makes the usage of wire for you so problematic.

Kind regards
Marco from the Wire Android Team

**Comment by agates** [comment]
@Gardosen thanks to the team for checking in on it.  So far the issue is better in 3.45.

**Comment by agates** [comment]
@Gardosen I hate to bring this to github but FYI zendesk is broken. They are not receiving my email replies to the support ticket.  I'm guessing it's because I host my own email/domain.

**Comment by neonpixii** [comment]
i am still having these sorts of issues; i think this issue should be re-opened because this is still a problem. wire constantly is sending messages to the wrong chat, forgetting what chat its in, words are on a lag, buttons are unresponsive, android contsantly asks if i wanna kill wire because it's running slowly, sometimes it will freeze on the loading screen.
